tests/constraints.in

```
@@ -1,5 +1,12 @@
 # This constraints file contains pins for the stable, tested versions of sphinx
 # and antsibull-docs that production builds rely upon.
+# This constraint file also pins other versions for which there are known limitations.
 
 sphinx == 7.2.5
 antsibull-docs == 2.16.2  # currently approved version
+
+sphinx-rtd-theme >= 2.0.0  # Fix 404 pages with new sphinx -- https://github.com/ansible/ansible-documentation/issues/678
+rstcheck < 6  # rstcheck 6.x has problem with rstcheck.core triggered by include files w/ sphinx directives https://github.com/rstcheck/rstcheck-core/issues/3
+jinja2 >= 3.0.0
+pyyaml >= 5.1
+resolvelib >= 0.5.3, < 1.1.0
```

I'd reference to https://github.com/ansible/ansible/blob/devel/requirements.txt for each of these:
```suggestion
jinja2 >= 3.0.0  # https://github.com/ansible/ansible/blob/devel/requirements.txt
pyyaml >= 5.1  # https://github.com/ansible/ansible/blob/devel/requirements.txt
resolvelib >= 0.5.3, < 1.1.0  # https://github.com/ansible/ansible/blob/devel/requirements.txt
```

---

Also the upper limit for resolvelib can be increased to 2.0.0. I'm not sure whether you want to do that in this PR or in a follow-up PR, though.

---

Maybe we should do the resolvelib change in a follow up.